cmake/onnxruntime_unittests.cmake

```diff
@@ -597,7 +601,10 @@ set(all_dependencies ${onnxruntime_test_providers_dependencies} )
   # so skip in this type of
   target_compile_definitions(onnxruntime_test_all PUBLIC -DSKIP_DEFAULT_LOGGER_TESTS)
   if (CMAKE_SYSTEM_NAME STREQUAL "iOS")
-      target_compile_definitions(onnxruntime_test_all_xc PUBLIC -DSKIP_DEFAULT_LOGGER_TESTS)
+    target_compile_definitions(onnxruntime_test_all_xc PUBLIC -DSKIP_DEFAULT_LOGGER_TESTS)
+  endif()
+  if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
```

Comment:
I think it is better to use check_cxx_compiler_flag instead
See: https://github.com/microsoft/onnxruntime/blob/master/cmake/CMakeLists.txt#L418

Reply:
Will update it. Thanks!

Reply:
I tried as below but got error: "cc1plus: error: unrecognized command line option '-Wno-unused-const-variable'". For gcc, check_cxx_compiler_flag doesn't works as expected for -Wno-unused-const-variable since it returns true but turns out gcc doesn't recognize this option.. According to gcc spec, suppressing this warning needs to be done by using variable attribute.

  check_cxx_compiler_flag(-Wno-unused-const-variable HAS_NO_UNUSED_CONST_VARIABLE)
  if (HAS_NO_UNUSED_CONST_VARIABLE)
    target_compile_options(onnxruntime_test_all PUBLIC "-Wno-unused-const-variable")
  endif()